### PR TITLE
Paging in BloodSugar history

### PR DESCRIPTION
**Pull request by msasikanth**
Implementing pagination support for `BloodSugarHistoryScreen`

https://www.pivotaltracker.com/story/show/172450143

### CODE REVIEW

**Comment by rakshakhegde** [comment]
@msasikanth There are 3 out of 6 PRs assigned to me. Can you please assign this to someone else?

**Comment by msasikanth** [comment]
> @msasikanth There are 3 out of 6 PRs assigned to me. Can you please assign this to someone else?

Reassigning to @vinaysshenoy

File: app/src/main/java/org/simple/clinic/bloodsugar/history/BloodSugarHistoryScreen.kt [hunk 1]
```diff
@@ -165,7 +158,19 @@ class BloodSugarHistoryScreen(
...
+  @SuppressLint("CheckResult")
+  override fun showBloodPressures(dataSourceFactory: BloodSugarHistoryListItemDataSourceFactory) {
+    val detaches = detaches()
+    // Initial load size hint should be a multiple of page size
+    val config = Config(
+        pageSize = 20,
+        prefetchDistance = 10,
+        initialLoadSizeHint = 40,
+        enablePlaceholders = false
+    )
```
Comment (vinaysshenoy) [inline app/src/main/java/org/simple/clinic/bloodsugar/history/BloodSugarHistoryScreen.kt:166]: If these values are the same for the ones we use to paginate the BP history screen, we should consider providing the values via a config object so they can be defined in one place.

File: app/src/main/java/org/simple/clinic/bloodsugar/history/BloodSugarHistoryScreenUiActions.kt [hunk 2]
```diff
@@ -1,9 +1,11 @@
 package org.simple.clinic.bloodsugar.history
 
+import org.simple.clinic.bloodsugar.BloodSugarHistoryListItemDataSourceFactory
 import org.simple.clinic.bloodsugar.BloodSugarMeasurement
 import java.util.UUID
 
 interface BloodSugarHistoryScreenUiActions {
   fun openBloodSugarEntrySheet(patientUuid: UUID)
   fun openBloodSugarUpdateSheet(measurement: BloodSugarMeasurement)
+  fun showBloodPressures(dataSourceFactory: BloodSugarHistoryListItemDataSourceFactory)
```
Comment (vinaysshenoy) [inline app/src/main/java/org/simple/clinic/bloodsugar/history/BloodSugarHistoryScreenUiActions.kt:10]: Shouldn't this be `showBloodSugars`?

File: app/src/test/java/org/simple/clinic/bloodsugar/history/BloodSugarHistoryScreenUpdateTest.kt [hunk 3]
```diff
@@ -63,19 +38,4 @@ class BloodSugarHistoryScreenUpdateTest {
             hasEffects(OpenBloodSugarEntrySheet(patientUuid) as BloodSugarHistoryScreenEffect)
         ))
   }
-
-  @Test
-  fun `when blood sugar is clicked, then open update sheet`() {
```
Comment (vinaysshenoy) [inline app/src/test/java/org/simple/clinic/bloodsugar/history/BloodSugarHistoryScreenUpdateTest.kt:68]: Why was this test removed?

Comment (msasikanth) [reply app/src/test/java/org/simple/clinic/bloodsugar/history/BloodSugarHistoryScreenUpdateTest.kt:68]: Accidental removal, Readding it.

File: app/src/main/java/org/simple/clinic/bloodsugar/history/BloodSugarHistoryScreen.kt (hunk 1 above)
Comment (msasikanth) [reply app/src/main/java/org/simple/clinic/bloodsugar/history/BloodSugarHistoryScreen.kt:166]: These values are the same for history screens only, should I create a pagination config for history screen only?

Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/bloodsugar/history/BloodSugarHistoryScreen.kt:166]: Yeah, we could do that.

Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/bloodsugar/history/BloodSugarHistoryScreen.kt:166]: Maybe something like `measurementHistoryPaginationConfig` or something.

**Pull request merged**